app/components/filter-plaques/component.js

```diff
@@ -1,21 +1,53 @@
 import Ember from 'ember';
 
-export default Ember.Component.extend({
+var ID_FILTERS = ['contributors', 'funders', 'publishers'];
 
+export default Ember.Component.extend({
+    filters: null,
     init () {
-        this._super(...arguments);
-        let parameters = this.get('parameters');
-        this.set('filters', Object.keys(parameters).map((key, idx, arr) => {
-            return {
-                "key": key,
-                "value": parameters[key]
-            }
-        }).filter((item) => {
-            if (item.key == "page") {
-                return false;
-            }
-            return true;
-        }));
+      this._super(...arguments);
+      let parameters = this.get('parameters');
+      var filters = Object.keys(parameters).filter((key) => {
+        return key !== "page";
+      }).map((key) => {
+        return {
+          "key": key,
+          "value": parameters[key]
+        }
+      });
+
+      // Find and format all query params that use ids for the value
+      var ids = filters.filter((item) => {
+          return ID_FILTERS.indexOf(item.key) !== -1;
+      }).map((param) => {
+          return {key: param.value}
+      });
+
+      // Fetch display names
+      if (ids.length > 0) {
+        this.fetchAgentDetails(ids).then((data) => {
+            var displayFilters = filters.map((filter) => {
+              var value = filter.value;
+              data.forEach((agentData) => {
+                if (value === agentData.id) {
+                  value = agentData.name;
```

Comment:
It seems unlikely that the name of a contributor/publisher/funder would be the id of another contributor/publisher/funder.